windows host: no mmap

Unfortunately mmap does not exist on Windows (yet) (Zig falsely has mmap in os.zig and the posix things did not get moved to posix.zig yet), so I get (even when targeting `windows-gnu`, but also for `windows-native`:
```
MadMax@DESKTOP-B9BTU5D MINGW64 ~/Desktop/ztags (master)
(ins)$ ../zig/bdeb/bin/zig.exe build -Drelease-safe -Dtarget=x86_64-windows-gnu
C:\Users\MadMax\Desktop\zig\bdeb\lib\zig\std\os.zig:100:23: error: root struct of file 'c' has no member named 'MAP'
pub const MAP = system.MAP;
                ~~~~~~^~~~
referenced by:
    findTags: src\Tags.zig:115:70
    main: src\main.zig:34:13
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

C:\Users\MadMax\Desktop\zig\bdeb\lib\zig\std\c.zig:146:122: error: root struct of file 'c' has no member named 'off_t'
pub extern "c" fn mmap(addr: ?*align(page_size) anyopaque, len: usize, prot: c_uint, flags: c_uint, fd: c.fd_t, offset: c.off_t) *anyopaque;
                                                                                                                        ~^~~~~~
error: ztags...
error: The following command exited with error code 1:
C:\Users\MadMax\Desktop\zig\bdeb\bin\zig.exe build-exe C:\Users\MadMax\Desktop\ztags\src\main.zig -OReleaseSafe --cache-dir C:\Users\MadMax\Desktop\ztags\zig-cache --global-cache-dir C:\Users\MadMax\AppData\Local\zig --name ztags -target x86_64-windows-gnu -mcpu x86_64 --enable-cache
error: the following build command failed with exit code 1:
C:\Users\MadMax\Desktop\ztags\zig-cache\o\28d96108f4f080e89942bee8a8b80c7c\build.exe C:\Users\MadMax\Desktop\zig\bdeb\bin\zig.exe C:\Users\MadMax\Desktop\ztags C:\Users\MadMax\Desktop\ztags\zig-cache C:\Users\MadMax\AppData\Local\zig -Drelease-safe -Dtarget=x86_64-windows-gnu
```

--- a/src/Tags.zig
+++ b/src/Tags.zig
@@ -88,6 +88,48 @@ pub fn deinit(self: *Tags) void {
     self.visited.deinit();
 }
 
+/// On Unix systems, memory map the given file. On Windows, just use read() for now (which
+/// allocates, and is slower, but is only temporary until a cross-platform "map file" function
+/// exists in the stdlib (or until someone implements it here)).
+fn mapFile(allocator: std.mem.Allocator, fname: []const u8) !?[:0]const u8 {
+    var file = try std.fs.cwd().openFile(fname, .{});
+    defer file.close();
+
+    const metadata = try file.metadata();
+    const size = metadata.size();
+    if (size == 0) {
+        return null;
+    }
+
+    if (metadata.kind() == .Directory) {
+        return error.NotFile;
+    }
+
+    switch (@import("builtin").os.tag) {
+        .windows => {
+            var array_list = try std.ArrayList(u8).initCapacity(allocator, size + 1);
+            defer array_list.deinit();
+            try file.reader().readAllArrayList(&array_list, size + 1);
+            return try array_list.toOwnedSliceSentinel(0);
+        },
+        else => {
+            var mapped = try std.os.mmap(null, size, std.os.PROT.READ, std.os.MAP.SHARED, file.handle, 0);
+            return std.meta.assumeSentinel(mapped, 0);
+        },
+    }
+}
+
+/// Unmap a file. On Windows, frees the allocated slice. See comments on mapFile.
+fn unmap(allocator: std.mem.Allocator, slice: [:0]const u8) void {
+    switch (@import("builtin").os.tag) {
+        .windows => allocator.free(slice),
+        else => {
+            const aligned = @alignCast(std.mem.page_size, slice);
+            std.os.munmap(aligned);
+        },
+    }
+}
+
 pub fn findTags(self: *Tags, fname: []const u8) anyerror!void {
     const gop = try self.visited.getOrPut(fname);
     if (gop.found_existing) {
@@ -95,25 +137,8 @@ pub fn findTags(self: *Tags, fname: []const u8) anyerror!void {
     }
     gop.value_ptr.* = {};
 
-    const mapped = a: {
-        var file = try std.fs.cwd().openFile(fname, .{});
-        defer file.close();
-
-        const metadata = try file.metadata();
-        const size = metadata.size();
-        if (size == 0) {
-            return;
-        }
-
-        if (metadata.kind() == .Directory) {
-            return error.NotFile;
-        }
-
-        break :a try std.os.mmap(null, size, std.os.PROT.READ, std.os.MAP.SHARED, file.handle, 0);
-    };
-    defer std.os.munmap(mapped);
-
-    const source = std.meta.assumeSentinel(mapped, 0);
+    const source = (try mapFile(self.allocator, fname)) orelse return;
+    defer unmap(self.allocator, source);
 
     var allocator = self.allocator;
     var ast = try std.zig.parse(allocator, source);

